inc/class-toolbar.php

```diff
@@ -106,48 +124,49 @@ public function action_admin_bar_menu( $wp_admin_bar ) {
 			}
 			if ( ! empty( $details_html ) ) {
 				$details_html = '<em>' . esc_html__( 'Environment Details', 'pantheon-hud' ) . '</em><br /> - ' . implode( '<br /> - ', $details_html );
-				$wp_admin_bar->add_node(
-					array(
-						'id'     => 'pantheon-hud-environment-details',
-						'parent' => 'pantheon-hud',
-						'title'  => $details_html,
-					)
-				);
+				$markup[]     = '<li id="wp-admin-bar-pantheon-hud-environment-details"><div class="ab-item ab-empty-item">' . $details_html . '</div></li>';
 			}
 		}
 
 		if ( $name && $env ) {
 			$wp_cli_stub = sprintf( 'terminus wp %s.%s', $name, $env );
-			$wp_admin_bar->add_node(
-				array(
-					'id'     => 'pantheon-hud-wp-cli-stub',
-					'parent' => 'pantheon-hud',
-					'title'  => '<em>' . esc_html__( 'WP-CLI via Terminus', 'pantheon-hud' ) . '</em><br /><input value="' . esc_attr( $wp_cli_stub ) . '">',
-				)
-			);
+			$markup[] = '<li id="wp-admin-bar-pantheon-hud-wp-cli-stub"><div class="ab-item ab-empty-item"><em>' . esc_html__( 'WP-CLI via Terminus', 'pantheon-hud' ) . '</em><br /><input value="' . esc_attr( $wp_cli_stub ) . '"></div></li>';
 		}
 
 		if ( $site_id && $env ) {
 			$dashboard_link = sprintf( 'https://dashboard.pantheon.io/sites/%s#%s/code', $site_id, $env );
-			$wp_admin_bar->add_node(
-				array(
-					'id'     => 'pantheon-hud-dashboard-link',
-					'parent' => 'pantheon-hud',
-					'href'   => $dashboard_link,
-					'title'  => esc_html__( 'Visit Pantheon Dashboard', 'pantheon-hud' ),
-					'meta'   => array(
-						'target' => '_blank',
-					),
-				)
-			);
+			$markup[] = sprintf( '<li id="wp-admin-bar-pantheon-hud-dashboard-link"><a class="ab-item" href="%s" target="_blank">Visit Pantheon Dashboard</a></li>', $dashboard_link );
 		}
 
+		$markup[] = '</ul>';
+		echo implode( PHP_EOL, $markup );
+		exit;
 	}
 
 	/**
 	 * Add admin bar inline styles.
 	 */
 	public function add_admin_bar_inline_styles() {
+		if ( ! is_admin_bar_showing() ) {
+			return;
+		}
+		$request_url = add_query_arg(
+			array(
+				'action'   => 'pantheon_hud_markup',
+				'_wpnonce' => wp_create_nonce( 'pantheon_hud' ),
+			),
+			admin_url( 'admin-ajax.php' )
+		);
+		$script      = <<<EOT
+jQuery('#wp-admin-bar-pantheon-hud').on('hover', function() {
+	if (jQuery('#wp-admin-bar-pantheon-hud-wp-admin-loading').length) {
+		jQuery.get('{$request_url}', function(data) {
+			jQuery('#wp-admin-bar-pantheon-hud .ab-sub-wrapper').html(data);
+		});
+	}
+});
+EOT;
```

Comment:
This presumes that jQuery is on the page, but this is not necessarily the case. In fact, Twenty Nineteen and Twenty Twenty don't use jQuery at all and so it is not enqueued on the page. Core recently improved its handling of the admin bar when jQuery is not present: https://core.trac.wordpress.org/ticket/47069

So this needs to be rewritten to use vanilla JS.

Reply:
@westonruter Great suggestion! updated with 5305ce1